tests/integration/targets/ec2_key/tasks/main.yml

```diff
@@ -345,6 +344,28 @@
           - 'result.key.name == "{{ec2_key_name}}"'
           - 'result.key.fingerprint == "{{fingerprint}}"'
 
+    # ============================================================
+    - name: test state=present with key_material (idempotency)
```

Comment:
We should also have some extra check_mode tests in here too to avoid regressions on #1367

Reply:
I was thinking the same but we don't currently have a way to guarantee that any new key pair has been created by our process when running integration testing